archinfo/arch_arm.py

```diff
@@ -195,9 +196,10 @@ def is_thumb(self, addr):  # pylint:disable=unused-argument
         br"\r\xc0\xa0\xe1\x04\xe0\x2d\xe5",                # mov r12, sp; push {lr}
     }
     thumb_prologs = {
-        br"[\x00-\xff]\xb5[\x00-\xff]\xb0",             # push {??, ??, ..., ??, lr}; sub sp, sp, #??
-        br"[\x00-\xff]\xb4\x00\xb5[\x00-\xff]\xb0",     # push {r?, r?}; push {lr}; sub sp, sp, #??
-        br"[\x00-\xff]\xb0[\x00-\xff]\x90",             # sub sp, sp, #??; str r0, [sp, ?]
+        br"[\x00-\xff]\xb5[\x80-\xff]\xb0",             # push {??, ??, ..., ??, lr}; sub sp, sp, #??
+        br"[\x80-\xff]\xb0\x80\xb4",                    # push {r7}; sub sp, sp, #??
```

Comment:
I believe what you want is `\x80\xb4\xb0[\x80-\xff]` since the first instruction is `push {r7}`.